sound/soc/sof/intel/hda-loader.c

```diff
@@ -279,10 +296,13 @@ int hda_cl_cleanup(struct device *dev, struct snd_dma_buffer *dmab,
 			  sd_offset + SOF_HDA_ADSP_REG_SD_BDLPU, 0);
 
 	snd_sof_dsp_write(sdev, HDA_DSP_HDA_BAR, sd_offset, 0);
-	snd_dma_free_pages(dmab);
-	dmab->area = NULL;
-	hstream->bufsize = 0;
-	hstream->format_val = 0;
+
+	if (!keep_buffer) {
```

Comment:
nit-pick: use same working instead of having 'keep_buffer' and 'persistent_buffer'. If it's the same thing use the same name, otherwise readers might think they are different concepts.